httpclient-backend/monix/src/main/scala/sttp/client/httpclient/monix/HttpClientMonixBackend.scala

```diff
@@ -0,0 +1,80 @@
+package sttp.client.httpclient.monix
+
+import java.io.InputStream
+import java.net.http.HttpRequest.BodyPublishers
+import java.net.http.{HttpClient, HttpRequest}
+import java.nio.ByteBuffer
+import java.util.concurrent.Flow
+
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.execution.cancelables.SingleAssignCancelable
+import monix.reactive.Observable
+import monix.reactive.observers.SafeSubscriber
+import sttp.client.httpclient.{HttpClientAsyncBackend, HttpClientBackend, WebSocketHandler}
+import sttp.client.impl.monix.TaskMonadAsyncError
+import sttp.client.{SttpBackend, _}
+
+import scala.util.{Success, Try}
+
+class HttpClientMonixBackend private (
+    client: HttpClient,
+    closeClient: Boolean,
+    customizeRequest: HttpRequest => HttpRequest
+)(implicit s: Scheduler)
+    extends HttpClientAsyncBackend[Task, Observable[ByteBuffer]](
+      client,
+      TaskMonadAsyncError,
+      closeClient,
+      customizeRequest
+    ) {
+
+  override def send[T](request: Request[T, Observable[ByteBuffer]]): Task[Response[T]] = {
+    super.send(request).guarantee(Task.shift)
```

Comment:
we should also shift after `openWebsocket`

maybe, in fact, this can be somehow extracted as a (stackable) trait?

Reply:
What do you mean by 'stackable' trait? I can see two options: either abstract class or composition. Obviously I prefer the second one.

Reply:
https://www.artima.com/scalazine/articles/stackable_trait_pattern.html

In other words, traits which have `abstract override` methods :)
Not a common Scala construction, but it's there

Reply:
But probably that's for another PR
Maybe you can raise an issue to unify that?

Reply:
sure, TIL